packages/flutter/test/widgets/automatic_keep_alive_test.dart

```diff
@@ -676,3 +688,27 @@ class _KeepAliveListenableLeakCheckerState extends State<KeepAliveListenableLeak
     return const Placeholder();
   }
 }
+
+class MultiKeepAliveNotificationDispatchWidget extends StatefulWidget {
+  const MultiKeepAliveNotificationDispatchWidget({super.key});
+
+  @override
+  State<MultiKeepAliveNotificationDispatchWidget> createState() => _MultiKeepAliveNotificationDispatchWidgetState();
+}
+
+class _MultiKeepAliveNotificationDispatchWidgetState extends State<MultiKeepAliveNotificationDispatchWidget> {
+  final KeepAliveHandle _handle = KeepAliveHandle();
+
+  @override
+  void deactivate() {
+    _handle.dispose();
+    super.deactivate();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    KeepAliveNotification(_handle).dispatch(context);
+    KeepAliveNotification(_handle).dispatch(context);
```

Comment:
This is just bad practice. Code shouldn't do this and we shouldn't encourage code that does it.

Reply:
More specifically: listeners to a change notifier aren't guaranteed to be cheap or idempotent.